modules/salt/files/bin/partial-reset-wiki.py

```diff
@@ -0,0 +1,82 @@
+import subprocess
+import sys
+import os
+import argparse
+import re
+import json
+from typing import Optional, TypedDict
+
+
+class DbClusterMap(TypedDict):
+    c1: str
+    c2: str
+    c3: str
+    c4: str
+    c5: str
+
+
+# Define the mapping of db clusters to db names
+db_clusters: DbClusterMap = {
+    'c1': 'db151',
+    'c2': 'db161',
+    'c3': 'db171',
+    'c4': 'db181',
+}
+
+
+def generate_salt_command(cluster: str, command: str) -> str:
+    return f'salt-ssh -E "{cluster}" cmd.run "{command}"'
+
+
+def execute_salt_command(salt_command: str, shell: bool = True, stdout: Optional[int] = None, text: Optional[bool] = None) -> Optional[subprocess.CompletedProcess]:
+    response = input(f'EXECUTE (type c(continue), s(kip), a(bort): {salt_command}')
+    if response in ['c', 'continue']:
+        return subprocess.run(salt_command, shell=shell, stdout=stdout, text=text)
+    if response in ['s', 'skip']:
+        return None
+    sys.exit(1)  # noqa: R503
+
+
+def get_db_cluster(wiki: str) -> str:
+    db_query = f'SELECT wiki_dbcluster FROM mhglobal.cw_wikis WHERE wiki_dbname = \\"{wiki}\\"'
+    command = generate_salt_command("db171", f"sudo -i mysql --skip-column-names -e '{db_query}'")
+    result = execute_salt_command(salt_command=command, stdout=subprocess.PIPE, text=True)
+    if result:
+         cluster_name = result.stdout.strip()
+         #print(cluster_name)
+         cluster_data = cluster_name.split('\n')
+         cluster_data_b = cluster_data[1].split(' ')
+         print(cluster_data_b)
+         #print("Extracted cluster_name:", cluster_name)  # Print cluster_name for debugging
+         cluster_name = cluster_data_b[4]
+
+         return db_clusters[cluster_name]  # type: ignore[literal-required]
```

Comment:
The additional comment misses the context completely because AI.

The risk of SQLi is low. Anyone running this has root anyone so input should be trusted.

Reply:
Up to you @Reception123